services/web/src/helpers/modal.js

```diff
@@ -0,0 +1,68 @@
+import React from 'react';
+import { Modal } from 'semantic';
+
+export default function modal(Component) {
+  class WrappedModal extends React.Component {
+    state = {
+      open: this.props.open,
+    };
+
+    componentDidUpdate(lastProps) {
+      const { open } = this.props;
+      if (open !== lastProps.open) {
+        this.setState({
+          open,
+        });
+      }
+    }
+
+    onClose = () => {
+      this.setState({ open: false });
+      if (this.props.onClose) this.props.onClose();
+    };
+
+    onOpen = () => {
+      this.setState({ open: true });
+      if (this.props.onOpen) this.props.onOpen();
+    };
+
+    render() {
+      const { open } = this.state;
+      const {
+        onClose,
+        onOpen,
+        closeIcon,
+        closeOnDimmerClick,
+        closeOnDocumentClick,
+        trigger,
+        centered,
+        size,
+        ...rest
+      } = this.props;
+
+      return (
+        <Modal
+          closeIcon={closeIcon}
+          closeOnDimmerClick={closeOnDimmerClick}
+          closeOnDocumentClick={closeOnDocumentClick}
+          size={size}
+          onOpen={this.onOpen}
+          onClose={this.onClose}
+          trigger={trigger}
+          centered={centered}
+          open={open}>
+          <Component {...rest} onClose={this.onClose} />
```

Comment:
I think this should be just `close` as that's how you're going to want to call it, no?

Reply:
i was ... but then i was thinking its better to keep the naming aligned with what you call it as a prop 
```
<EditUser onClose={...} />
```

So it aligns with any prob you might set via props, instead of making a special case out it. Let me know

Reply:
The semantic modal prop is almost mis-named I think... it should be something more like `onCloseTriggered` or `onCloseClick`...  maybe think of it as a separate method:

```
close = () => {
  this.onClose();
}
```

Reply:
done